### DISCUSSION TITLE
Favorite Packages list

### DISCUSSION THREAD

**Discussion by aim97** [npm/feedback#185] [Website] [answered]
Wouldn't be helpful to allow everyone to keep a list of his favorite packages, I'm new to NPM, Many packages are used in every app, I also find a lot of interesting packages on NPM site but may not have a need to use them now, and every time I see one, I automatically search for that subscribe or favorite button in the page.

So, I was thinking it would be helpful to have such a feature that allows me to keep a list of the packages I prefer, or find useful, and have that list in the right place (where the packages are, on NPM site).

**Comment by ljharb** [comment]
Many years ago, npm promised a public “curated collections”-like feature, where users could make multiple collections of packages and would be able to follow those of others, but it never materialized.

This would be a great feature to help with package discovery, and to avoid the inherent potential problems in npm altering its own search algorithm over time to effectively privilege some packages over others.

**Comment by MylesBorins** [comment, marked as answer]
Hey @aim97... fun story, we already kinda have this feature

You can do `npm star [package]` to star a package ([doc](https://docs.npmjs.com/cli/v6/commands/npm-star))
You can do `npm stars [user]` to view their stars ([doc](https://docs.npmjs.com/cli/v6/commands/npm-stars/))

Although we obviously do not have a UI view of this on the website. I think that we need to do some thinking into how to make this information useful... a collection of packages in a stack is a more more interesting story then a list of unrelated packages.

**Reply by ljharb** [reply]
A historical note: the npm website used to surface this information, but stopped doing so a number of years ago. I believe there was even a few years in between where `npm star` simply failed.

**Reply by aim97** [reply]
Thanks a lot for pointing that out, I'll start using it from now on.

And about how to make that more useful, I was thinking 🤔 of allowing users to make collections of related packages, ones that are usually used together by that user. 
Then he can use `npm install [collection]` to install all packages of that collection in one go. 

But I'm satisfied enough with `npm star` for now, thanks a lot. 😊

**Reply by lucianodiisouza** [reply]
wow! I was arrived here, just searching something like `mark an npm package as favorite`, the idea of install a collection of npm packages, seems interesting, for me personally, help a lot, cause im a frontend teacher and i make the same installs an every class that starts the sames modules. gj.

**Reply by artu-hnrq** [reply]
Nice to know that is possible to `npm star` packages through CLI 
Anyway, it still would be a pretty useful UI feature, right?
I'll wait for it

**Reply by ovanderzee** [reply]
You can install multiple packages in one go by just summing the packages in a space separated list:
```bash
npm install package1 package2 package3 package4 package5
```

**Reply by aim97** [reply]
@ovanderzee this way you have to remember the names of all those packages, but by adding them to a collection it makes it easier to install them all at once without much thinking trying to remember each single one. 
It just feels convenient this way.
